addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1462,10 +1462,11 @@ var SnippetsMenu = Widget.extend({
             const range = selection.getRangeAt(0);
             $(range.startContainer).closest('.o_default_snippet_text').removeClass('o_default_snippet_text');
         });
-        const refreshSnippetEditors = _.debounce(() => {
+        const refreshSnippetEditors = _.debounce(async () => {
             for (const snippetEditor of this.snippetEditors) {
                 this._mutex.exec(() => snippetEditor.destroy());
             }
+            await this._updateInvisibleDOM();
```

Comment:
Putting this between the snippetEditor destroy and the comment about it is confusing :grimacing: In the end I would put this as the very end of this function and also add a blank line above it as it's unrelated to the rest. I would also not await it: _updateInvisibleDOM uses the mutex internally + debouncing an async method... strange.

```javascript
        [...]
        this._activateSnippet($(target));
    }

    this._updateInvisibleDOM();
}
```

Reply:
Done. It's ready for review :)